examples/dreambooth/train_dreambooth_lora.py

```diff
@@ -75,6 +75,13 @@
 logger = get_logger(__name__)
 
 
+def free_memory():
+    if torch.cuda.is_available():
+        torch.cuda.empty_cache()
+    if hasattr(torch, "xpu") and torch.xpu.is_available():
+        torch.xpu.empty_cache()
```

Comment:
Let's directly update this function no?
https://github.com/huggingface/diffusers/blob/4ace7d0483d9b5016575a2d51119c33172c005ed/src/diffusers/training_utils.py#L290